### PR TITLE
Adsk Contrib - Add Core Renderer Built-in Config

### PR DESCRIPTION
**Pull request by cozdas**
- Adding "core-renderer-config-v1.0.0-rc1" as one of the built-in configs

- Added a preprocessor macro definition to the OpenColorABI indicating the feature set this OCIO library is compiled with. One of these macros will be defined: OCIO_FEATURESET_FULL, OCIO_FEATURESET_NANO and OCIO_FEATURESET_CUSTOM
- Pulling the common code that all built-in configs use into a separate cpp file for deduplicaton. Added BuiltinConfigUtils.h/cpp for this.
- Minor code cleanup
- Added tests for the new built-in CoreRenderer Config.

### CODE REVIEW

File: src/OpenColorIO/builtinconfigs/BuiltinConfigRegistry.cpp [hunk 1]
```diff
@@ -15,19 +15,26 @@
...
+#include "builtinconfigs/CoreRendererConfig.h"
 
 static constexpr char OUT_OF_RANGE_EXCEPTION_TEXT[] = "Config index is out of range.";
 
 // TODO: Remove once getDefaultBuiltinConfigName is removed.
+#if defined(OCIO_FEATURESET_NANO)
 static constexpr char DEFAULT_BUILTIN_CONFIG[] = "cg-config-v2.1.0_aces-v1.3_ocio-v2.3";
+#else
+static constexpr char DEFAULT_BUILTIN_CONFIG[] = "ocio://core-renderer-config-v1.0.0-rc1";
+#endif
```
Comment (doug-walker) [inline src/OpenColorIO/builtinconfigs/BuiltinConfigRegistry.cpp:27]: We don't need to change the default for nano.

**Review by doug-walker — COMMENTED** [review]
I'll check more closely tomorrow, but this looks like it's on the right track!

File: include/OpenColorIO/OpenColorABI.h.in [hunk 2]
```diff
@@ -26,6 +26,11 @@
 #define OCIO_VERSION_MAJOR @OpenColorIO_VERSION_MAJOR@
 #define OCIO_VERSION_MINOR @OpenColorIO_VERSION_MINOR@
 
+// The feature set this OCIO library is compiled with
+#cmakedefine OCIO_FEATURESET_FULL
+#cmakedefine OCIO_FEATURESET_NANO
+#cmakedefine OCIO_FEATURESET_CUSTOM
+
```
Comment (cozdas) [inline include/OpenColorIO/OpenColorABI.h.in:33]: only one of them will be defined. Example output:

```
// The feature set this OCIO library is compiled with
#define OCIO_FEATURESET_FULL
/* #undef OCIO_FEATURESET_NANO */
/* #undef OCIO_FEATURESET_CUSTOM */
```

File: src/OpenColorIO/CMakeLists.txt [hunk 3]
```diff
@@ -11,8 +11,10 @@ set(SOURCES
     BakingUtils.cpp
     BitDepthUtils.cpp
     builtinconfigs/BuiltinConfigRegistry.cpp
+    builtinconfigs/BuiltinConfigUtils.cpp
```
Comment (cozdas) [inline src/OpenColorIO/CMakeLists.txt:14]: Created a new cpp/h file holding the common code for all built-in config generation. We may add more functions to it in the future.

File: src/OpenColorIO/builtinconfigs/CoreRendererConfig.cpp [hunk 4]
```diff
@@ -0,0 +1,567 @@
...
+ConstConfigRcPtr Create_CORE_RENDERER_CONFIG_V100_RC1();
+
+void Register(BuiltinConfigRegistryImpl & registry) noexcept
+{
+    // If a new built-in config is added, do not forget to update the LATEST_CORE_RENDERER_BUILTIN_CONFIG_URI
+    // variable (in BuiltinConfigRegistry.cpp).
+
+    registry.addBuiltin(
+        "core-renderer-config-v1.0.0-rc1",
+        "Core Renderer Config",
```
Comment (cozdas) [inline src/OpenColorIO/builtinconfigs/CoreRendererConfig.cpp:27]: Probably needs a better description/name here. Please let me know.

File: src/OpenColorIO/builtinconfigs/BuiltinConfigUtils.cpp [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+// Copyright Contributors to the OpenColorIO Project.
+
+#include "BuiltinConfigUtils.h"
+
+namespace OCIO_NAMESPACE
+{
+    namespace BUILTIN_CONFIG_UTILS
+    {
+
+    void AddColorSpace(ConfigRcPtr& cfg, const char* name, ReferenceSpaceType type, const char** aliases, BitDepth bitDepth, const char** categories, const char* encoding, const char* eqGroup, const char* family, bool isData, ConstTransformRcPtr trFrom, ConstTransformRcPtr trTo, const char* desc)
```
Comment (doug-walker) [inline src/OpenColorIO/builtinconfigs/BuiltinConfigUtils.cpp:11]: Minor: Please wrap to a reasonable length.  We try to keep lines between about 80-100 characters, typically.

File: src/OpenColorIO/builtinconfigs/CoreRendererConfig.cpp (hunk 4 above)
Comment (doug-walker) [reply src/OpenColorIO/builtinconfigs/CoreRendererConfig.cpp:27]: I'll suggest:
"ASWF Color Interop Forum -- Color Space Encodings for Texture Assets and CG Rendering"

**Review by doug-walker — APPROVED** [review]
LGTM!

**Comment by doug-walker** [comment]
Actually, when I re-enabled showing white space differences, there still are some.  Please revert those.

**Comment by cozdas** [comment]
Those white space differences you see are the tabs -which leaked into the nanocolor branch in the previous commits- now replaced by the spaces. This PR doesn't introduce new tabs but fixes existing tabs with spaces.

File: src/OpenColorIO/builtinconfigs/BuiltinConfigUtils.cpp (hunk 5 above)
Comment (cozdas) [reply src/OpenColorIO/builtinconfigs/BuiltinConfigUtils.cpp:11]: Good point. This is my IDE messing around with my formatting upon paste. I think in the long run we need to incorporate a clang format file into our repo, formalizing these unwritten rules/preferences.

**Review by doug-walker — APPROVED** [review]
Looks good!

**Pull request merged**